.pre-commit-config.yaml

```diff
@@ -32,7 +32,7 @@ repos:
     -   id: end-of-file-fixer           # Makes sure files end in a newline and only a newline.
     # -   id: fix-encoding-pragma         # Add # -*- coding: utf-8 -*- to the top of python files.
     # -   id: file-contents-sorter        # Sort the lines in specified files (defaults to alphabetical). You must provide list of target files as input to it.
-    -   id: flake8                      # Run flake8 on your python files.
```

Comment:
We should run pylint and YAPF here and get rid of the individual scripts

https://github.com/pre-commit/mirrors-pylint
https://github.com/pre-commit/mirrors-yapf

Reply:
I'm not sure about the accuracy of these two, but at least for flake8, I remember that pre-commit's flake8 would pass my code while our flake8 script would fail me. Maybe pre-commit is not always up to date?

Reply:
Our flake8 script has special rules about files which are existing/added/changed which is why we won't use pre-commit for it (we could write a custom pre-commit script for it, but probably not worth it right now).

YAPF and pylint are straightforward invocations of the tool using our setup.cfg. It should be fine.

Angel did the original configuration and can tell you all about it.